lib/Fetcher/FeedFetcher.php

```diff
@@ -294,17 +306,51 @@ protected function buildItem(ItemInterface $parsedItem, ?string $body = null, bo
         return $item;
     }
 
+    /**
+     * Return the favicon for a given feed and url
+     *
+     * @param FeedInterface $feed     Feed to check for a logo
+     * @param string        $url      Original URL for the feed
+     *
+     * @return string|mixed|bool
+     */
+    private function getFavicon(FeedInterface $feed, string $url)
+    {
+        $favicon = $feed->getLogo();
+
+        // check if feed has a logo
+        if (!is_null($favicon)) {
+            $favicon_path = join("/", [$this->ITempManager->getTempBaseDir(), basename($favicon)]);
+            file_put_contents(
+                $favicon_path,
+                file_get_contents($url)
+            );
+            $is_image = substr(mime_content_type($favicon_path), 0, 5) === "image";
+            // check if file is actually an image
+            if ($is_image) {
+                list($width, $height, $type, $attr) = getimagesize($favicon_path);
+                // check if image is square else fall back to favicon
+                if ($width !== $height) {
+                    $favicon = $this->faviconFactory->get($url);
+                }
+            }
+        } else {
+            $favicon = $this->faviconFactory->get($url);
+        }
```

Comment:
This greatly reduces cyclomatic complexity (and improves readability)

```suggestion
        // check if feed has a logo
        if (is_null($favicon)) {
            return $this->faviconFactory->get($url);
        }
        
        $favicon_path = join("/", [$this->ITempManager->getTempBaseDir(), basename($favicon)]);
        file_put_contents(
            $favicon_path,
            file_get_contents($url)
        );

        $is_image = substr(mime_content_type($favicon_path), 0, 5) === "image";
        // check if file is actually an image
        if (!$is_image) {
            return $this->faviconFactory->get($url);
        }

        list($width, $height, $type, $attr) = getimagesize($favicon_path);
        // check if image is square else fall back to favicon
        if ($width !== $height) {
            return $this->faviconFactory->get($url);
        }
        
        return $favicon;
```